lib/src/bootloader.rs

```diff
@@ -12,15 +16,44 @@ pub(crate) fn install_via_bootupd(
     device: &Utf8Path,
     rootfs: &Utf8Path,
     configopts: &crate::install::InstallConfigOpts,
+    via_loopback: bool,
 ) -> Result<()> {
+    let device_str: &str;
+    let prepboot_dir;
     let verbose = std::env::var_os("BOOTC_BOOTLOADER_DEBUG").map(|_| "-vvvv");
     // bootc defaults to only targeting the platform boot method.
     let bootupd_opts = (!configopts.generic_image).then_some(["--update-firmware", "--auto"]);
+
+    if cfg!(target_arch = "powerpc64") {
+        // get PowerPC-PReP-boot device information
+        if via_loopback {
+            // While writing to a qcow2 image use first partition
+            // of loopback device as it points to
+            // PowerPC-PReP-boot partition
+            device_str = "/dev/loop0p1";
+        } else {
+            // While writing to a disk use realpath to extract
+            // PowerPC-PReP-boot partition
+            let result = Command::new("realpath")
```

Comment:
I don't think this will do much different from the builtin https://doc.rust-lang.org/std/fs/fn.canonicalize.html

Reply:
Thanks, this helps.